Server upgrade

TswusayonaTsamsiyu opened the pull request:
# Summary
Upgraded the server Node version and all packages to their latest versions, then fixed breaking changes.

# Notes:
- Changed `start:prod` script to depend on global `pm2` install (we should make sure it works during deployment)
- Commits https://github.com/TheResonator/resonators-server/pull/37/commits/5c0cfc218e0f808f89642431b7693715bf632d63 and https://github.com/TheResonator/resonators-server/pull/37/commits/16e753317d74426668c51bb72a4c190cf4b67227 require the following SQL commands to be executed on the operational DB after deploy:
```
ALTER TABLE leader_clinics RENAME COLUMN "isPrimary" TO "is_primary";
ALTER TABLE leader_clinics RENAME COLUMN "isLeaderAccepted" TO "is_leader_accepted";
ALTER TABLE google_accounts RENAME COLUMN "createdAt" TO "created_at";
ALTER TABLE google_accounts RENAME COLUMN "updatedAt" TO "updated_at";
```
- We must also merge this: https://github.com/TheResonator/resonators-client/pull/30

sOfekS reviewed (commented):
I don't get it. Why did you change some columns from camelCase to snake_case and some the other way around? Besides, I think we shouldn't change column names just for the convention, it's unfortunate they were made against convention but it could only lead to issues and not really necessary right now. Or am I missing something?

TswusayonaTsamsiyu commented on src/application/login.js:
```diff
@@ -1,5 +1,5 @@
 import {user_logins as UserLogin} from '../db/sequelize/models';
-import uuid from 'uuid/v4';
+import { v4 as uuid } from "uuid";
```
The reason it's 2 different ways is I started manually changing them to the `uuid.v4()` version, but after a few files realized I could do a project-wide search-and-replace with this `v4 as uuid` version without breaking the code.
I don't see why this is so critical to maintain consistency on this. It's not like there's a new UUID version every month...

sOfekS replied:
It's just weird that you're using 2 different methods for this even if they're practically the same. It can cause confusion that's all, why not be consistent with it?

TswusayonaTsamsiyu replied:
Just because it's annoying manual work. There are like 90 places to change.

TswusayonaTsamsiyu commented:
Regarding your general question about changing column names:
`sequelize` has the `underscored` option for defining models, which turns all camelCase field names to snake_case when querying the DB, and back to camelCase when receiving query results. All of our models use this option. It seems though that something in `sequelize`'s implementation has been changed between versions 3 and 6, which caused the following problems:
- `leader_clinics` for some reason had 2 field names in camelCase, AND they were camelCase in the DB, so seems like someone in the past really didn't care about conforming to project convention. Somehow in the previous version `sequelize` still used the camelCased DB columns (maybe it tried both camelCase and snake_case), but in the current version the `underscored` option strictly converts all field names to snake_case before queries, and so it tried accessing non-existent columns. The `RENAME` queries change the columns in the DB to fix this problem, and then I decided to also rename the fields in the code according to the convention.
- The second problem is because the timestamp fields (`createdAt` & `updatedAt`) are implicitly defined on our models using `sequelize`'s `timestamps` option (which is set to `true` by default). Since they are not explicitly defined in our code in snake_case, they are only accessible in their camelCase names, according to the `underscored` mechanism. I don't know why it worked in the previous version, but now it doesn't. Maybe the entire `underscored` implementation changed. That's why I've changed them to camelCase everywhere in the code. The only issue is that 2 models for some reason had these timestamp fields explicitly defined, and one of them (`google_accounts`) used camelCase versions, AND the columns in the DB table were camelCase as well, so this is the reason for the second `RENAME` query.

BTW the entire code IMO should be changed to camelCase. This is the popular JS convention, and it will also reduce confusion. If you look at the objects `sequelize` now returns as query results, you'll see they contain both camelCase attributes, converted by the `underscored` mechanism, and also snake_case duplicates, which is just silly.
